### PR TITLE
Fix #226 and upgrade dependencies

### LINKED ISSUE #226: [BUG] 部分设备上无法选中标签
**复现步骤**
复现该 Bug 的步骤：
1. 到树洞编辑器页面
2. 点击 Tag，随便选择若干 Tag
3. 继续选择时，突然无法选中列表项。表现为点击屏幕任意处，列表都会关闭

**截图**
<video src="https://user-images.githubusercontent.com/8389787/227709886-9f5d0c25-5b33-49ed-b660-711831edb54f.mp4"/>

**系统信息**
 - 设备型号：n/a
 - 系统版本：Android
 - 旦夕版本（见「设置-关于本应用」）：1.3.11

### PR DESCRIPTION
**Pull request by w568w**
This PR fixes #226 and upgrade dependencies.

I also applied `dart fix` to adapt to the recently introduced `super.xxx` grammar.

**Since I have changed some key dependencies and NOT walked through a thorough test, please check**:

- Whether saving to gallery still works on mobile platforms (i.e. Android, iOS)
- Whether the storage location (or keys?) of shared preferences have changed (there is a way to check this: upgrade from previous build, finding that all settings and accounts are lost, then the shared preferences should be read from different location and the answer is YES)

### CODE REVIEW

**Comment by w568w** [comment]
@singularity-s0 @ivanfei-1 @Boreas618 Bump it

**Comment by ivanfei-1** [comment]
I will take a look soon.

**Comment by ivanfei-1** [comment]
> This PR fixes #226 and upgrade dependencies.
> 
> I also applied `dart fix` to adapt to the recently introduced `super.xxx` grammar.
> 
> **Since I have changed some key dependencies and NOT walked through a thorough test, please check**:
> 
> * Whether saving to gallery still works on mobile platforms (i.e. Android, iOS)
> * Whether the storage location (or keys?) of shared preferences have changed (there is a way to check this: upgrade from previous build, finding that all settings and accounts are lost, then the shared preferences should be read from different location and the answer is YES)

1. Yes. The "Save to Gallery" feature is still useable on iOS devices. I haven't checked Android Devices though I think it should be useable too.
2. Yes. Both the account and settings information have been changed.

File: pubspec.yaml [hunk 1]
```diff
@@ -115,34 +111,22 @@ dependency_overrides:
...
-  bitsdojo_window_macos:
-    git:
-      url: https://github.com/DartGit-dev/bitsdojo_window.git
-      path: bitsdojo_window_macos
-      ref: master
-  bitsdojo_window_linux:
+
+  # Temporary fixed version of bitsdojo_window_linux_v3.
+  # Should be removed if the fix is merged in the upstream.
+  bitsdojo_window_linux_v3:
```
Comment (ivanfei-1) [inline pubspec.yaml:117]: Should this be fixed before merge?

**Comment by w568w** [comment]
> Yes. Both the account and settings information have been changed.

That sounds bad. We (of course) hope that the storage location of shared preferences have **NOT** changed, or we must migrate from previous versions.

File: pubspec.yaml (hunk 1 above)
Comment (w568w) [reply pubspec.yaml:117]: No we don't need to wait. Or if you want, check <https://github.com/DartGit-dev/bitsdojo_window/pull/1>.

**Comment by w568w** [comment]
Progress report: I will divide this PR into several smaller PRs considering that the changes blocking this PR have nothing to do with #226. Close it for now.

**Pull request closed without merging**